src/features/swap/hooks/useIsExpiredSwap.ts

```diff
@@ -12,13 +9,33 @@ const INTERVAL_IN_MS = 10_000
  */
 const useIsExpiredSwap = (txInfo: TransactionInfo) => {
   const [isExpired, setIsExpired] = useState<boolean>(false)
-  const [counter] = useIntervalCounter(INTERVAL_IN_MS)
+  const timerRef = useRef<NodeJS.Timeout | null>(null)
 
   useEffect(() => {
     if (!isSwapOrderTxInfo(txInfo)) return
 
-    setIsExpired(Date.now() > txInfo.validUntil * 1000)
-  }, [counter, txInfo])
```

Comment:
This updates on every `counter` increase but will say `true` at some point so `useIsExpiredSwap` would then always return the same boolean value. How is this causing rerenders?

Reply:
Thanks for looking into this! Makes a lot more sense to only check this once if we have the exact timestamp. No need to do this every 10s with a poll.